ui/components/multichain/edit-accounts-modal/edit-accounts-modal.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  EthAccountType,
+  InternalAccount,
+  KeyringAccountType,
+} from '@metamask/keyring-api';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  getInternalAccounts,
+  getUpdatedAndSortedAccounts,
+} from '../../../selectors';
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  Checkbox,
+  Box,
+} from '../../component-library';
+import { AccountListItem } from '..';
+import { MergedInternalAccount } from '../../../selectors/selectors.types';
+import { mergeAccounts } from '../account-list-menu/account-list-menu';
+
+type EditAccountsModalProps = {
+  onClose: () => void;
+  allowedAccountTypes?: KeyringAccountType[]; // Made optional to match default
+};
+
+const defaultAllowedAccountTypes: KeyringAccountType[] = [
+  EthAccountType.Eoa,
+  EthAccountType.Erc4337,
+];
+
+export const EditAccountsModal = ({
+  onClose,
+  allowedAccountTypes = defaultAllowedAccountTypes,
+}: EditAccountsModalProps) => {
+  const t = useI18nContext();
+  const accounts = useSelector(getUpdatedAndSortedAccounts);
+  const internalAccounts = useSelector(getInternalAccounts);
+
+  const mergedAccounts: MergedInternalAccount[] = useMemo(() => {
+    return mergeAccounts(accounts, internalAccounts).filter(
+      (account: InternalAccount) => allowedAccountTypes.includes(account.type),
+    );
+  }, [accounts, internalAccounts, allowedAccountTypes]); // Add allowedAccountTypes to dependency array
+
+  return (
+    <Modal
+      isOpen
+      onClose={onClose} // Simplified inline function
+      data-testid="edit-accounts-modal"
+      className="edit-accounts-modal"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader onClose={onClose}>{t('editAccounts')}</ModalHeader>
+        <Box padding={4}>
+          <Checkbox
+            label={t('selectAll')}
+            isChecked
+            gap={4}
+            // Uncomment and implement these if needed:
+            // onClick={() => (allAreSelected() ? deselectAll() : selectAll())}
+            // isIndeterminate={isIndeterminate}
+          />
+        </Box>
+        {mergedAccounts.map((account) => (
+          <AccountListItem
+            account={account}
+            key={account.address}
+            isPinned={Boolean(account.pinned)}
+            startAccessory={<Checkbox isChecked />}
+            onClick={() => console.log(null)}
+            selected={false}
```

Comment:
Please finish implementing this

Reply:
nit: might be nice to consider including the avatar with badge for the account, as well as the avatargroup with mock values, that would get the UI done, this was useful for mobile since it revealed some adjustments DS team had to do to components, maybe your extension components are already matching figma

![2024-08-28 12 56 55](https://github.com/user-attachments/assets/2d059a39-02ff-4c12-b169-06f31dbb7402)

Reply:
It's a dumb component so having a console on click is fine. The selection is not gonna be based on this component so it's fine to keep it this way. @jonybur

Reply:
@EtherWizard33 the connected badges for avatar is visible when it's in the popup view. So, it's already included in there